hippod/report_generator.py

```diff
@@ -83,70 +83,108 @@ def store_data(app, data, sub_dir):
                       separators=(',', ': '))
-                    dst.write(data)
+                    dst.write(content)
                 return dst_path
             else:
                 return None
 
 
-        def add_data(description_path, file_path):
+        def store_attachment(self, app, attachment_path, sub_dir):
+            with open(attachment_path, 'r') as attach:
+                content = json.load(attach)
+            if not os.path.isdir(sub_dir):
+                os.mkdir(sub_dir)
+            dst_path = os.path.join(sub_dir, 'attachment.db')
+            with open(dst_path, 'w') as dst:
+                content = json.dumps(content, sort_keys=True,indent=4,
+                      separators=(',', ': '))
+                dst.write(content)
+            return dst_path
+
+
+        def get_achievement_content(self, achievement_path):
+            with open(achievement_path) as achievement:
+                content = json.load(achievement)
+                return content
+
+
+        def get_attachment_content(self, attachment_path):
+            with open(attachment_path) as attach:
+                content = json.load(attach)
+                return content
+
+
+        def add_data(self, description_path, file_path):
             with open(description_path, 'r') as text:
                 description = text.read()
             with open(description_path, 'w') as text:
                 description = str(description) + '\n' + '![image caption here]({})'.format(file_path)
                 text.write(description)
 
-        def add_achievement(description_path, achievement_path, title):
+
+        def add_achievement(self, description_path, achievement_path, title, achievement_data, attachment_path):
+            attach_content = self.get_attachment_content(attachment_path)
+            achievement_content = self.get_achievement_content(achievement_path)
             if description_path == None:
                 # remove '/achievement.db' of the path and create a 'description.md' file in this directory
                 tmp_item_path = os.path.dirname(achievement_path)
                 description_path = os.path.join(tmp_item_path, 'description.md')
-                with open(achievement_path, 'r') as achievement:
-                    content = json.load(achievement)
-                    result = content['result']
-                with open(description_path, 'w') as text:
-                    description = '''
-# {} #
-
------------------------------------------------------------------------------------
-**Test Result:**    {}
 
------------------------------------------------------------------------------------
+                result = achievement_content['result']
+                submitter = achievement_content['submitter']
+                test_date = achievement_content['test-date']
 
-                    '''.format(title, result)
+                with open(description_path, 'w') as text:
```

Comment:
don't call it "text", better file, or fh for file handler or fd for file descriptor.